### PR TITLE
Alipay payment check

### PR DESCRIPTION
**Pull request by jigar-f**
This pull request improves the reliability and accuracy of user account status updates after a purchase, especially for asynchronous payment flows (like Alipay or Stripe). It introduces a more robust mechanism to detect when a purchase or renewal has actually been credited to the user’s account, and ensures that the UI is updated promptly when new user data is available from the backend.

The most important changes are:

**Backend and Core Logic Improvements:**

* Added retry logic with backoff to the Go backend for fetching user data at startup, and introduced a new `user-data` event to notify the frontend when fresh user data is available. [[1]](diffhunk://#diff-62f35878f2bc3e51174e5c90579a40e547ce99ff63b5001d2d60d5186f726a12R60-R75) [[2]](diffhunk://#diff-62f35878f2bc3e51174e5c90579a40e547ce99ff63b5001d2d60d5186f726a12L321-R366)
* Dart now listens for the `user-data` event and reloads cached user data when notified, ensuring the UI reflects backend changes immediately. [[1]](diffhunk://#diff-a796ab2befef63f34821d469e984851e9db569f8a428836a363eb49d0244918cR99-R102) [[2]](diffhunk://#diff-b459e7f66927a6d26fe94728c4dd77b971b3c9bb52e8f82d4692f86ead45233eR54-R69)

**Purchase Confirmation and Account Status Logic:**

* Refactored purchase confirmation logic to distinguish between new purchases and renewals by checking whether the user’s expiration has advanced past its previous value, not just whether they are “pro.” Added configurable retry schedules for polling the backend after a purchase.
* Updated payment flow to handle cases where the payment webview is closed without a completion redirect: the backend is polled to check for a credited purchase, and a negative result is treated as a cancellation, not an error. [[1]](diffhunk://#diff-df1e00ef79211512b982a3f87f2ee298f997256d63bc1cf5085622bec568bb29R349-R375) [[2]](diffhunk://#diff-df1e00ef79211512b982a3f87f2ee298f997256d63bc1cf5085622bec568bb29R433-R460) [[3]](diffhunk://#diff-df1e00ef79211512b982a3f87f2ee298f997256d63bc1cf5085622bec568bb29R487-R524)

**Testing:**

* Added unit tests for the new purchase detection logic to ensure correct behavior for first-time purchases, renewals, and edge cases.

## Summary by CodeRabbit

* **New Features**
  * Purchase confirmation now verifies account status and subscription expiration before reporting success.
  * Payment flows verify purchases when checkout closes unexpectedly.
  * Account data refreshes automatically after server synchronization.
  * Startup account-data updates retry when delayed.
  * Added support for viewing, refreshing, acknowledging, and managing user messages.

* **Bug Fixes**
  * Unchanged subscription expirations are no longer treated as completed renewals.
  * Payment redirects are cleared when verification fails.

### CODE REVIEW

File: lantern-core/core.go [hunk 1]
```diff
@@ -319,19 +331,39 @@ func (lc *LanternCore) notifyFlutter(event EventType, message string) {
...
 	}
-	if _, err := lc.client.FetchUserData(lc.ctx); err != nil {
-		slog.Error("Startup user-data fetch failed", "error", err)
-		return
+	attempts := len(startupUserDataFetchDelays) + 1
+	for i := 0; i < attempts; i++ {
+		_, err := lc.client.FetchUserData(lc.ctx)
+		if err == nil {
+			slog.Debug("Startup user-data fetch succeeded", "userID", userID, "attempt", i+1)
+			lc.notifyFlutter(EventTypeUserData, "")
```
[the remark this answers is not in the document]
Comment (jigar-f) [reply lantern-core/core.go:362]: That's not possible. That method is synchronous.

File: lib/features/auth/choose_payment_method.dart [hunk 2]
```diff
@@ -461,23 +508,45 @@ class ChoosePaymentMethod extends HookConsumerWidget {
...
-    if (isPro) {
+    if (purchaseConfirmed) {
       ref.read(paymentSessionProvider.notifier).clearRedirect();
       resolveRoute(context);
+    } else if (unconfirmedClose) {
+      appLogger.info(
+        'Webview closed without completion redirect and backend shows no '
+        'new purchase; treating as cancel',
+      );
+      ref.read(paymentSessionProvider.notifier).clearRedirect();
```
[the remark this answers is not in the document]
Comment (jigar-f) [reply lib/features/auth/choose_payment_method.dart:531]: The checkout guard is paymentRedirectInFlight (cleared in `finally` on every path), so this branch doesn't enable a duplicate checkout. paymentSessionProvider is only a signup back-press guard, not an idempotency check; if the payment settles later, core's user-data event grants Pro regardless. Leaving as is.

**Review by atavism — APPROVED** [review]
LGTM! Just a couple suggestions here https://github.com/getlantern/lantern/pull/9037
